mkosi/__init__.py

```diff
@@ -6329,7 +6329,7 @@ def xescape(s: str) -> str:
 
 def build_auxiliary_output_path(args: Union[argparse.Namespace, MkosiArgs], suffix: str, can_compress: bool = False) -> Path:
     output = strip_suffixes(args.output)
-    compression = f".{should_compress_output(args)}" if can_compress else ''
+    compression = f".{should_compress_output(args)}" if can_compress and should_compress_output(args) else ''
```

Comment:
Just to make this a bit more manageable.
```suggestion
    should_compress = should_compress_output(args)
    compression = f".{should_compress}" if can_compress and should_compress else ''
```